Service/File/ScpFileUploadService.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Garlic\Bus\Service\File;
+
+use Garlic\Bus\Exceptions\FileUploadException;
+
+/**
+ * Class ScpFileUploadService
+ *
+ * @package Garlic\Bus\Service\File
+ */
+class ScpFileUploadService
+{
+    /**
+     * @var string
+     */
+    private $username;
+    /**
+     * @var string
+     */
+    private $password;
+    /**
+     * @var string
+     */
+    private $uploadDir;
+
+    /**
+     * ScpFileUploadService constructor.
+     *
+     * @param $username
+     * @param $password
+     * @param $uploadDir
+     */
+    public function __construct($username, $password, $uploadDir)
+    {
+        $this->username = $username;
+        $this->password = $password;
+        $this->uploadDir = $uploadDir;
+    }
+
+    /**
+     * @param array $metadata
+     *
+     * @return string
+     * @throws FileUploadException
+     */
+    public function getFile(array $metadata)
+    {
+        $connection = $this->getSshConnection($metadata);
+        $filePath = $this->uploadDir . md5(time() . $metadata['origin_name']);
+        ssh2_scp_recv($connection, $metadata['path'], $filePath);
```

Comment:
potential non-FileUploadException exception. wrap into try/catch for proper handling?

Reply:
done